### PR TITLE
Flow-366: c2 : list envelope documents

### PR DESCRIPTION
**Pull request by harshitav-docusign**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/DocuSignDemo/script.csx [hunk 1]
```diff
@@ -1739,6 +1746,30 @@ public class Script : ScriptBase
...
+      JObject newBody = new JObject();
+
+      if (body["envelopeDocuments"] == null)
+      {
+        throw new ConnectorException(HttpStatusCode.BadRequest, "ValidationFailure: Envelope do not contain documents");
+      }
+
+      foreach(var document in body["envelopeDocuments"] as JArray ?? new JArray())
+      {
+        envelopeDocuments.Add(new JObject(
```
Comment (harshitav-docusign) [inline certified-connectors/DocuSignDemo/script.csx]: The shorthand is working only when JObject has a reference defined like this:
var workflowObj = new JObject()
        {
          ["type"] = id["workflowId"],
          ["name"] = id["defaultName"]
        };
       I will leave the code the way it is. If I give a reference name I have to change the schema.

File: certified-connectors/DocuSignDemo/apiDefinition.swagger.json [hunk 2]
```diff
@@ -2193,6 +2193,59 @@
...
       }
     },
+    "/accounts/{accountId}/envelopes/{envelopeId}/envelopeDocuments": {
+      "get": {
+        "tags": [
+          "DocuSign"
+        ],
+        "summary": "List envelope documents",
+        "description": "List documents in an envelope.",
+        "operationId": "ListEnvelopeDocuments",
```
Comment (harshitav-docusign) [inline certified-connectors/DocuSignDemo/apiDefinition.swagger.json:2203]: Updated the operationId to "ListEnvelopeDocuments"

File: certified-connectors/DocuSignDemo/script.csx (hunk 1 above)
Comment (sriharip-docusign) [reply certified-connectors/DocuSignDemo/script.csx]: If you do something like the following, you don't need to change any schema and still use the shorthand.
```javascript
envelopeDocuments.Add(new JObject()
{
   ["documentId"] = document["documentId"],
    ["name"] = document["name"]
}
```

Comment (harshitav-docusign) [reply certified-connectors/DocuSignDemo/script.csx]: That worked great! Thank you.

File: certified-connectors/DocuSignDemo/script.csx [hunk 3]
```diff
@@ -1739,6 +1746,31 @@ public class Script : ScriptBase
...
+    if ("ListEnvelopeDocuments".Equals(this.Context.OperationId, StringComparison.OrdinalIgnoreCase))
+    {
+      var body = ParseContentAsJObject(await response.Content.ReadAsStringAsync().ConfigureAwait(false), false);
+      var query = HttpUtility.ParseQueryString(this.Context.Request.RequestUri.Query);
+      JArray envelopeDocuments = new JArray();
+      JObject newBody = new JObject();
+
+      if (body["envelopeDocuments"] == null)
+      {
+        throw new ConnectorException(HttpStatusCode.BadRequest, "ValidationFailure: Envelope do not contain documents");
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSignDemo/script.csx:1758]: nit: Envelope does not contain any documents

**Review by sriharip-docusign — APPROVED** [review]

**Pull request merged**